src/units/abilities.cpp

```diff
@@ -471,6 +471,55 @@ bool unit::has_ability_type(const std::string& ability) const
 	return !abilities_.child_range(ability).empty();
 }
 
+//this string abilitie is used for purpose of add a second halo if condition is matched.
+//It is put here header declaration commodity.
+const std::string unit::halo_abilities() const
+{
+	unit_ability_list halo_list = get_abilities("halo");
+	if(!halo_list.empty()){
+		utils::erase_if(halo_list, [&](const unit_ability& i) {
+			return ((*i.ability_cfg)["image"].empty() || (*i.ability_cfg)["image"].str().empty());
```

Comment:
I think this was needed if any ability could add a halo, but if only tagname `[halo]` abilities can add a halo then it doesn't seem necessary to check for the empty string. A `[halo]image=""` could be a way to disable other `[halo]` abilities.

Reply:
checking empty string removed

Reply:
Much more can be removed - the whole `utils::erase_if` block is unneeded, AFAICS.